src/vector/platform/WebPlatform.ts

```diff
@@ -184,17 +183,8 @@ export default class WebPlatform extends VectorBasePlatform {
 
     getDefaultDeviceDisplayName(): string {
         // strip query-string and fragment from uri
-        const u = url.parse(window.location.href);
-        u.protocol = "";
-        u.search = "";
-        u.hash = "";
-        // Remove trailing slash if present
-        u.pathname = u.pathname.replace(/\/$/, "");
-
-        let appName = u.format();
-        // Remove leading slashes if present
-        appName = appName.replace(/^\/\//, "");
-        // `appName` is now in the format `develop.element.io`.
+        const url = new URL(window.location.href);
+        const appName = url.host; // `appName` in the format `develop.element.io`
```

Comment:
If you wonder why this got so simple I put the explanation below (as image):

(please click to open & scale)
![image](https://user-images.githubusercontent.com/4066435/107656717-7f80b380-6c95-11eb-827d-d7864c31a60d.png)